frontend/packages/ux-editor/src/components/leftMenu/ConfigureLayoutSetPanel.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { ChangeEvent, useEffect, useState, useRef, MouseEvent } from 'react';
+import { useTranslation, Trans } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import classes from "./LeftMenu.module.css";
+import { useConfigureLayoutSetMutation } from "../../hooks/mutations/useConfigureLayoutSetMutation";
+import { Button, ButtonVariant, TextField  } from '@digdir/design-system-react';
+import { Popover } from '@mui/material';
+import { InformationIcon } from '@navikt/aksel-icons';
+import { altinnDocsUrl } from 'app-shared/utils/urlHelper';
+
+export const ConfigureLayoutSetPanel = () => {
+  const { org, app } = useParams();
+  const { t } = useTranslation();
+  const configureLayoutSetMutation = useConfigureLayoutSetMutation(org, app);
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+  const [popoverOpen, setPopoverOpen] = useState<boolean>(false);
+  const [layoutSetName, setLayoutSetName] = useState<string>('');
+  const [editLayoutSetName, setEditLayoutSetName] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const configPanelRef = useRef<HTMLDivElement>(null);
+
+  function handleConfigureLayoutSet() {
+    if (layoutSetName === '') {
+      setErrorMessage(t('left_menu.pages_error_empty'));
+    }
+    else {
+      configureLayoutSetMutation.mutate({ layoutSetName });
+    }
+  }
+
+  const handlePopoverOpen = (event: MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+    setPopoverOpen(true);
+  };
+
+  const handlePopoverClose = () => {
+    setAnchorEl(null);
+    setPopoverOpen(false);
+  };
+
+  const handleKeyPress = async (event: any) => {
+    if (event.key === 'Enter') {
+      handleConfigureLayoutSet()
+      setEditLayoutSetName(false);
+    } else if (event.key === 'Escape') {
+      setEditLayoutSetName(false);
+      setLayoutSetName('');
+    }
+  };
+
+  const handleClickOutside = (event: any) => {
+    if (configPanelRef.current && !configPanelRef.current.contains(event.target as Node)) {
+      setEditLayoutSetName(false);
+      setLayoutSetName('');
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  const handleConfigureLayoutSetButtonClick = () => {
+    setEditLayoutSetName(!editLayoutSetName);
+  }
+
+  const handleOnChange = (event: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+    const nameRegex = new RegExp('^[a-zA-Z0-9_\\-\\.]*$');
```

Comment:
I Javascript kan man skrive regex slik:
```suggestion
    const nameRegex = /^[a-zA-Z0-9_\\-\\.]*$/;
```
Foreslår for øvrig å flytte denne valideringen ut i en egen funksjon, med en test.